File: src/main/java/site/yapp/study/todolist/common/response/ErrorCode.java
package site.yapp.study.todolist.common.response;

import lombok.AccessLevel;
import lombok.Getter;
import lombok.RequiredArgsConstructor;
import org.springframework.http.HttpStatus;

import static org.springframework.http.HttpStatus.*;

@Getter
@RequiredArgsConstructor(access = AccessLevel.PROTECTED)
public enum ErrorCode {
    /* 400 BAD REQUEST */
    FAILED_BULK_TODO(BAD_REQUEST, "벌크 할일 생성 실패"),

    /* 404 NOT FOUND */
    NOT_FOUND_TODO(NOT_FOUND, "존재하지 않는 할일입니다."),
    DELETED_TODO(NOT_FOUND, "삭제된 할일입니다.");

    private final HttpStatus status;
    private final String message;

}

File: src/main/kotlin/com/example/studytodolist/common/exception/ErrorCode.kt
package com.example.studytodolist.common.exception

import org.springframework.http.HttpStatus

enum class ErrorCode(val status: HttpStatus, val code: Int, val message: String) {
    /**
     * 200X
     * Todo 관련 에러
     */
    TODO_NOT_FOUND(HttpStatus.NOT_FOUND, 2000, "해당하는 Todo가 존재하지 않습니다."),
    TODO_BULK_INSERT_FAILED(HttpStatus.BAD_REQUEST, 2001, "TODO Bulk 삽입에 실패했습니다.")
}